dataflow/src/main/java/org/checkerframework/dataflow/expression/JavaExpression.java

```diff
@@ -914,4 +916,85 @@ private static boolean isVarargsInvocation(
     return TypesUtils.getArrayDepth(ElementUtils.getType(lastParamElt))
         != TypesUtils.getArrayDepth(lastArgType);
   }
+
+  /**
+   * Find the declaration of the receiver of a method call in a given method tree.
+   *
+   * <p>The receiver should appear in one of two places, either as a formal parameter to the method,
+   * or as a local variable.
+   *
+   * @param tree the method tree
+   * @param receiver the receiver for which to look up a declaration
+   * @return the declaration of the receiver of the method call, if found. Otherwise, null
+   */
+  public static @Nullable VariableTree getReceiverDeclarationInMethod(
+      @Nullable MethodTree tree, JavaExpression receiver) {
+    if (tree == null) {
+      return null;
+    }
+    List<? extends VariableTree> params = tree.getParameters();
+    for (VariableTree param : params) {
+      if (param.getName().toString().equals(receiver.toString())) {
+        return param;
+      }
+    }
+    for (StatementTree statement : tree.getBody().getStatements()) {
+      if (statement instanceof VariableTree) {
+        VariableTree localVariableTree = (VariableTree) statement;
+        if (localVariableTree.getName().toString().equals(receiver.toString())) {
```

Comment:
How does this handle the possibility that a variable declaration declares two variables, as in 
```int x, y, z;```
?

Reply:
I have a test case here, which behaves as expected
```java
import java.util.Optional;
import java.util.List;
import java.util.ArrayList;
import java.util.stream.*;

import org.checkerframework.checker.nonempty.qual.*;

class Main {

  void foo() {
    @NonEmpty List<Integer> foo = List.of(1), bar = List.of(2);
    List<Integer> baz = new ArrayList<>();
    foo.stream().findAny().get(); // OK
    bar.stream().findAny().get(); // OK

    baz.stream().findAny().get(); // Error
  }
}
```